### ISSUE TITLE
Type alias of member types of union type get discarded

### ISSUE THREAD

**Issue by malinthar** [ballerina-platform/ballerina-lang#34300]
**Description:**
Consider the following scenario.
```ballerina
type Type1 string | int;
type Type2 string;


public function main() {
      Type1 | Type2 |int | mod1:MyType myvar;
      if myvar is 
}
```
Here the member types of myvar is as follows.

![memeber types](https://user-images.githubusercontent.com/35211477/145808174-72253637-a2df-450b-af6d-ff94dc225d39.png)

In the following image the Type1 has expanded to string,int members. Due to this we do not recognize Type1 as a subtype of the type desc of myvar. 

**Steps to reproduce:**
Mentioned above

**Affected Versions:**
slbeta7-snapshot

**Comment by pubudu91** [comment]
@malinthar Please use `userSpecifiedMemberTypes()` if you need the members as how the user specified it in code. `memberTypeDescriptors()` give the expanded set of members.

**Comment by malinthar** [comment]
> @malinthar Please use `userSpecifiedMemberTypes()` if you need the members as how the user specified it in code. `memberTypeDescriptors()` give the expanded set of members.

Type 2 is not expanded in the above case. Isn't that inconsistent? Or does the expansion only happen for user specified union types?

**Comment by pubudu91** [comment]
> > @malinthar Please use `userSpecifiedMemberTypes()` if you need the members as how the user specified it in code. `memberTypeDescriptors()` give the expanded set of members.
> 
> Type2 is not expanded in the above case. Isn't that inconsistent? Or does the expansion only happen for user specified union types?

Yes, I guess that should've been expanded too 🤔 It looks like non-union type refs aren't expanded when adding to this member set in the compiler. We plan on going through the type ref implementation again. Will see about this as well with that effort. 

For your case, `userSpecifiedMemberTypes()` would work right IINM?

**Comment by malinthar** [comment]
> > > @malinthar Please use `userSpecifiedMemberTypes()` if you need the members as how the user specified it in code. `memberTypeDescriptors()` give the expanded set of members.
> > 
> > 
> > Type2 is not expanded in the above case. Isn't that inconsistent? Or does the expansion only happen for user specified union types?
> 
> Yes, I guess that should've been expanded too thinking It looks like non-union type refs aren't expanded when adding to this member set in the compiler. We plan on going through the type ref implementation again. Will see about this as well with that effort.
> 
> For your case, `userSpecifiedMemberTypes()` would work right IINM?

Yes that'll do.

**Comment by dulajdilshan** [comment]
The issues seem to be fixed. Hence closing this

1. This issue mentioned in the description can be sone using `userSpecifiedMemberTypes()` method of the `UnionTypeSymbol` class

<img width="896" alt="image" src="https://user-images.githubusercontent.com/25485997/199672760-038b7376-2f86-47db-8af2-fcd6fbaae119.png">

2. **TypeRefType symbols not getting expanded in union-type** also seems to be fixed in 2201.2.0+ (tested on the `2201.2.0` ) 
<img width="894" alt="image" src="https://user-images.githubusercontent.com/25485997/199672421-5cd7201e-11fc-45cc-8b72-210183797945.png">
